src/graph/sampler.cc

```diff
@@ -1032,8 +1052,30 @@ NegSubgraph NegEdgeSubgraph(GraphPtr gptr, IdArray relations, const Subgraph &po
       neg_vids.push_back(i);
       neg_map[i] = i;
     }
+
+    // Get all nodes in the positive side.
+    for (int64_t i = 0; i < num_pos_edges; i++) {
+      dgl_id_t vid = induced_vid_data[unchanged[i]];
+      local_pos_vids.push_back(neg_map[vid]);
+    }
+    // There is no guarantee that the nodes in the vector are unique.
+    std::sort(local_pos_vids.begin(), local_pos_vids.end());
+    auto it = std::unique(local_pos_vids.begin(), local_pos_vids.end());
+    local_pos_vids.resize(it - local_pos_vids.begin());
```

Comment:
My general question is how will this change if a bipartite graph is returned? Will the logic become cleaner?

Reply:
yes, the logic will be cleaner if it's a bipartite graph.